app/components/img-selector.js

```diff
@@ -0,0 +1,32 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    thumbnail: null,
+    actions: {
+        uploadImage: function(e) {
+            var self = this;
+
+            //var canvas = this.$().find('.img-selector-canvas')[0];
+            //var ctx = canvas.getContext('2d');
+
+            var reader = new window.FileReader();
+            reader.onload = function(event){
+                /*
+                var img = new Image();
+                img.onload = function(){
+                    canvas.width = img.width;
+                    canvas.height = img.height;
+                    ctx.drawImage(img,0,0);
+                };
+                img.src = event.target.result;
+                 */
+                self.set('thumbnail', event.target.result);
+                var onSetImage = self.get('onSetImage');
```

Comment:
I can't find good examples of `sendAction` in the latest Ember docs. Instead I see:

https://guides.emberjs.com/v2.3.0/components/triggering-changes-with-actions/#toc_passing-the-action-to-the-component